python/ray/_private/gcs_aio_client.py

```diff
@@ -1,26 +1,7 @@
-import os
 import logging
-from functools import partial
-from typing import Dict, List, Optional
-from concurrent.futures import ThreadPoolExecutor
-from ray._raylet import GcsClient, NewGcsClient, JobID
-from ray.core.generated import (
-    gcs_pb2,
-)
-import ray._private.utils
-from ray._private.ray_constants import env_integer
+from typing import Optional
 import ray
-
-# Number of executor threads. No more than this number of concurrent GcsAioClient calls
-# can happen. Extra requests will need to wait for the existing requests to finish.
-# Executor rules:
-# If the arg `executor` in GcsAioClient constructor is set, use it.
-# Otherwise if env var `GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT` is set, use it.
-# Otherwise, use 5.
-# This is only used for the OldGcsAioClient.
-GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT = env_integer(
-    "GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT", 5
-)
+from ray._raylet import NewGcsClient
```

Comment:
NewGcsClient -> GcsClient?

Reply:
Yeah problem is we have `NewGcsClient` the binding and the `GcsClient` the one with call frequency counter. Not sure about the naming here...

Reply:
Got it. 

Should it be `GcsClient`, `GcsSyncClient` and `GcsAsyncClient`?

Reply:
There are ~150 `GcsClient` needs to be renamed to GcsSyncClient, which is too big a change. Instead, I propose:

- GcsClient: sync
- GcsAioClient: async
- InnerGcsClient: binding for both

and it's apparent users should not use InnerGcsClient.